sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ObjectsShouldNotBeDisposedMoreThanOnce.cs

```diff
@@ -142,10 +142,36 @@ public override ProgramState PreProcessUsingStatement(ProgramPoint programPoint,
 
             public override ProgramState PreProcessInstruction(ProgramPoint programPoint, ProgramState programState)
             {
+                var instruction = programPoint.Block.Instructions[programPoint.Offset];
+                if (instruction is InvocationExpressionSyntax invocation)
+                {
+                    return VisitInvocationExpression(invocation, programState);
+                }
+                else if (instruction is VariableDeclaratorSyntax declarator &&
+                    declarator.Parent?.Parent is LocalDeclarationStatementSyntax declaration &&
+                    declaration.UsingKeyword() != null &&
+                    declaration.UsingKeyword().Kind() != SyntaxKind.None)
```

Comment:
`declaration.UsingKeyword() != null` is always true since the SyntaxToken is a struct. Could we use `declaration.IsKind(SyntaxKind.UsingKeyword)` instead?